zerver/tornado/event_queue.py

```diff
@@ -1149,11 +1184,22 @@ def process_message_update_event(
     )
     stream_push_user_ids = set(event_template.pop("stream_push_user_ids", []))
     stream_email_user_ids = set(event_template.pop("stream_email_user_ids", []))
-    wildcard_mention_user_ids = set(event_template.pop("wildcard_mention_user_ids", []))
+
+    # TODO/compatibility: Translation code for the rename of
+    # `wildcard_mention_user_ids` to `stream_wildcard_mention_user_ids`.
+    # Remove this when one can no longer directly upgrade from 7.x to main.
```

Comment:
[As discussed earlier](https://chat.zulip.org/#narrow/stream/378-api-design/topic/allowed.20visibility.20policies/near/1580224):
>   I don't think it's necessarily a problem to have the API allow you to follow topics -- it may be useful for testing parts of the feature on [chat.zulip.org](http://chat.zulip.org/) before we're ready to turn it on for everyone, and I don't think we intend to have intermediate states where it is a security problem that you were allowed to follow a topic.

> We just don't guarantee to users that it'll be a good experience to do so.

So, should we care about providing a good experience?  Our docs don't have "Follow" as a visibility policy as of now.

Reply:
Makes sense. We just need to make sure the server doesn't throw errors because of this (`.get` without a default value, or `.pop` can throw errors if the key isn't present in the dict, so we need to make sure isn't any such code).
Users not receiving notifications for followed topics is fine, but runtime errors aren't

Reply:
We have default value available.